components/common/link/NavigationLink.tsx

```diff
@@ -0,0 +1,28 @@
+import type { FC } from 'react';
+
+import NextLink from 'next/link';
+import { useRouter } from 'next/router';
+
+import MuiLink, { LinkProps } from '@mui/material/Link';
+
+const NavigationLink: FC<LinkProps> = props => {
+  const { variant, href, children, ...restProps } = props;
+
+  const router = useRouter();
+  const isActive = router.asPath === href;
```

Comment:
I am unsure if it's the right approach to check the link to be active. According to the doc, `asPath` is the path as shown in the browser, including the **search params** and respecting the `trailingSlash` configuration. `basePath` and `locale` are not included.

We will have two variants of links:
- Dynamic route will be matched via `props.as` - case studies, blog
- Static route will be matched via `props.href` - everything else

Also, we need to check whether the **router fields are updated client-side**.

Please, take a look at [this example](https://github.com/vercel/next.js/blob/canary/examples/active-class-name/components/ActiveLink.tsx) by Next.js Team

Reply:
'+